phoenix-core/src/main/java/org/apache/phoenix/mapreduce/ViewTTLDeleteJobMapper.java

```diff
@@ -0,0 +1,163 @@
+    private static final Logger LOGGER = LoggerFactory.getLogger(ViewTTLDeleteJobMapper.class);
+    private MultiViewJobStatusTracker multiViewJobStatusTracker;
+
+    private void initMultiViewJobStatusTracker(Configuration config) {
+        try {
+            Class<?> defaultViewDeletionTrackerClass = DefaultMultiViewJobStatusTracker.class;
+            if (config.get(PhoenixConfigurationUtil.MAPREDUCE_VIEW_TTL_MAPPER_TRACKER_CLAZZ) != null) {
+                LOGGER.info("Using customized tracker class : " + config.get(PhoenixConfigurationUtil.MAPREDUCE_VIEW_TTL_MAPPER_TRACKER_CLAZZ));
+                defaultViewDeletionTrackerClass = Class.forName(
+                        config.get(PhoenixConfigurationUtil.MAPREDUCE_VIEW_TTL_MAPPER_TRACKER_CLAZZ));
+            } else {
+                LOGGER.info("Using default tracker class ");
+            }
+            this.multiViewJobStatusTracker = (MultiViewJobStatusTracker) defaultViewDeletionTrackerClass.newInstance();
+        } catch (Exception e) {
+            LOGGER.info("exception " + e.getMessage());
+            LOGGER.info("stack trace" + e.getStackTrace().toString());
+        }
+    }
+
+    @Override
+    protected void map(NullWritable key, ViewInfoTracker value,
+                       Context context) {
+        final Configuration config = context.getConfiguration();
+
+        if (this.multiViewJobStatusTracker == null) {
+            initMultiViewJobStatusTracker(config);
+        }
+
+        LOGGER.debug(String.format("Deleting from view %s, TenantID %s, and TTL value: %d",
+                value.getViewName(), value.getTenantId(), value.getViewTtl()));
+
+        try (PhoenixConnection connection =(PhoenixConnection) ConnectionUtil.getInputConnection(config) ){
+            if (value.getTenantId() != null && !value.getTenantId().equals("NULL")) {
+                try (PhoenixConnection tenantConnection = (PhoenixConnection)PhoenixViewTtlUtil.
+                        buildTenantConnection(connection.getURL(), value.getTenantId())) {
+                    deletingExpiredRows(tenantConnection, value, config);
+                }
+            } else {
+                deletingExpiredRows(connection, value, config);
+            }
+
+        } catch (SQLException e) {
+            LOGGER.error(e.getErrorCode() + e.getSQLState(), e.getStackTrace());
+        }
+    }
+
+    private void deletingExpiredRows(PhoenixConnection connection, ViewInfoTracker value, Configuration config)
+            throws SQLException {
+        PTable view = PhoenixRuntime.getTable(connection, value.getViewName());
+
+        String deleteIfExpiredStatement = "SELECT /*+ NO_INDEX */ count(*) FROM " + value.getViewName();
+        deletingExpiredRows(connection, view, Long.valueOf(value.getViewTtl()),
+                deleteIfExpiredStatement, config);
+        List<PTable> allIndexesOnView = view.getIndexes();
```

Comment:
AFAIK, this also gets indexes on the base table. Is it expected?

Reply:
that's expected. We delete expired rows for the base table, as well as the index table.

Reply:
@yanxinyi We are only deleting from view indexes right?

Reply:
yes. that's correct.